pkg/apis/options/options.go

```diff
@@ -222,6 +223,7 @@ func NewFlagSet() *pflag.FlagSet {
 
 	flagSet.StringSlice("email-domain", []string{}, "authenticate emails with the specified domain (may be given multiple times). Use * to authenticate any email")
 	flagSet.StringSlice("whitelist-domain", []string{}, "allowed domains for redirection after authentication. Prefix domain with a . to allow subdomains (eg .example.com)")
+	flagSet.StringSlice("extra-header", []string{}, "extra headers to add to the upstream request.")
```

Comment:
Bear in mind that a string slice will split on commas so users can do `--extra-header "foo: bar, baz: bar"` and that would give `[]string{"foo: bar", "baz: bar"}`, but `--extra-header "foo: bar,baz"` comes out as `[]string{"foo: bar", "baz"}` which may not have been the intention, perhaps we need to document this and/or handle it?

Reply:
uh.... yeah... http headers could contain comma's. Does the arg parsing have an escape for comma's?

Reply:
you could use `StringArray` instead of `StringSlice`: https://godoc.org/github.com/spf13/pflag#StringArray

Reply:
Done.